lib/screens/main_screen/main_screen.dart

```diff
@@ -28,17 +35,51 @@ class MainScreen extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'PICOS',
-      localizationsDelegates: AppLocalizations.localizationsDelegates,
-      supportedLocales: AppLocalizations.supportedLocales,
-      routes: <String, WidgetBuilder> {
-        '/questionaire': (BuildContext context) => const QuestionaireScreen()
-      },
-      theme: ThemeData(
-        primarySwatch: Colors.blue,
+    final MedicationsRepository medicationsRepository =
+        MedicationsRepository(medicationsApi: LocalStorageMedicationsApi());
+
+    return MultiRepositoryProvider(
+      providers: <RepositoryProvider<MedicationsRepository>>[
+        RepositoryProvider<MedicationsRepository>.value(
+          value: medicationsRepository,
+        )
+      ],
+      child: MultiBlocProvider(
+        providers: <BlocProvider<MedicationsListBloc>>[
+          BlocProvider<MedicationsListBloc>(
+            create: (BuildContext context) => MedicationsListBloc(
+              medicationsRepository: context.read<MedicationsRepository>(),
+            )..add(const MedicationsListSubscriptionRequested()),
+          ),
+        ],
+        child: MaterialApp(
+            title: 'PICOS',
+            localizationsDelegates: AppLocalizations.localizationsDelegates,
+            supportedLocales: AppLocalizations.supportedLocales,
+            theme: ThemeData(
+              primaryColor: Colors.blue,
+              primarySwatch: Colors.blue,
+              appBarTheme: const AppBarTheme(
+                backgroundColor: Colors.blue,
+              ),
+              backgroundColor: const Color(0xFFF2F2F2),
+              scaffoldBackgroundColor: const Color(0xFFF2F2F2),
+              shadowColor: Colors.grey,
+              textButtonTheme: TextButtonThemeData(
+                style: ButtonStyle(
+                  foregroundColor:
+                      MaterialStateProperty.all<Color>(Colors.white),
+                ),
+              ),
+            ),
+            home: const BottomBar(title: 'PICOS'),
+            routes: <String, Widget Function(BuildContext)>{
+              '/my-medications': (BuildContext ctx) =>
+                  const MyMedicationsScreen(),
+              '/add-medication': (BuildContext ctx) =>
+                  const AddMedicationScreen()
+            }),
```

Comment:
Please split here }), is better to have ")," in a new line and at the same level as the corosponding "(" in line 55